exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -207,6 +210,21 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
+   * If cluster version is the same as default, update option to current drillbit version.
+   */
+  private void checkAndUpdateClusterVersionOption(OptionManager optionManager, DrillbitEndpoint drillbitEndpoint) {
+    OptionValue versionOption = optionManager.getOption(ExecConstants.CLUSTER_VERSION);
+    OptionValidator validator = SystemOptionManager.getValidator(ExecConstants.CLUSTER_VERSION);
+    if (versionOption.equals(validator.getDefault())) {
+      optionManager.setOption(OptionValue.createOption(
```

Comment:
I'm not convinced that having the admin specify the cluster option is the right approach. The proper approach is for each Drillbit to detect that its own version conflicts with one or more of the others. But, for this to work, Drillbits have to advertise their version. Do we have that ability?